### PR TITLE
UCX/PML/SPML/OSC: fix compiler warnings

### LINKED ISSUE #9984: Warnings on master branch: ucx related
Building on CentOS with gcc 8.5.0 compiler:
```c
common_ucx.c: In function ‘opal_common_ucx_check_device’:
common_ucx.c:212:31: warning: ISO C does not support the 'm' scanf flag [-Wformat=]
     ret = sscanf(device_name, "%m[^:]%*d", &ib_device_name);
                               ^~~~~~~~~~~
common_ucx.c: In function ‘opal_common_ucx_support_level’:
common_ucx.c:308:22: warning: ISO C does not support the 'm' scanf flag [-Wformat=]
                      "# resource %*d : md %*d dev %*d flags -- %m[^/ \n\r]/%m[^/ \n\r]",
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
common_ucx.c:308:22: warning: ISO C does not support the 'm' scanf flag [-Wformat=]
btl_uct_component.c: In function ‘mca_btl_uct_component_open’:
btl_uct_component.c:127:24: warning: comparison of integer expressions of different signedness: ‘int’ and ‘uint32_t’ {aka ‘unsigned int’} [-Wsign-compare]
         if (core_count <= opal_process_info.num_local_peers || !opal_using_threads()) {
                        ^~
btl_uct_component.c: In function ‘mca_btl_uct_component_process_uct_component’:
btl_uct_component.c:459:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘unsigned int’ [-Wsign-compare]
     for (int i = 0; i < attr.md_resource_count; ++i) {
                       ^
btl_uct_component.c: In function ‘mca_btl_uct_component_init’:
btl_uct_component.c:488:14: warning: unused variable ‘resource_count’ [-Wunused-variable]
     unsigned resource_count;
              ^~~~~~~~~~~~~~
btl_uct_component.c:487:29: warning: unused variable ‘resources’ [-Wunused-variable]
     uct_md_resource_desc_t *resources;
                             ^~~~~~~~~
osc_ucx_passive_target.c: In function ‘ompi_osc_ucx_unlock_all’:
osc_ucx_passive_target.c:229:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘uint64_t’ {aka ‘volatile long unsigned int’} [-Wsign-compare]
         for (i = 0; i < module->state.dynamic_win_count; i++) {
                       ^
osc_ucx_passive_target.c:244:13: warning: declaration of ‘i’ shadows a previous local [-Wshadow]
         int i;
             ^
osc_ucx_passive_target.c:220:9: note: shadowed declaration is here
         i = 0, ret = OMPI_SUCCESS;
         ^
pml_ucx.c: In function ‘mca_pml_ucx_enable’:
pml_ucx.c:500:21: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
                     (MPI_Type_internal_copy_attr_function*)MPI_TYPE_NULL_COPY_FN;
                     ^
pml_ucx.c:499:36: warning: assignment to ‘MPI_Type_internal_copy_attr_function’ {aka ‘int (*)(struct ompi_datatype_t *, int,  void *, void *, void *, int *, struct ompi_datatype_t *)’} from incompatible pointer type ‘int (**)(struct ompi_datatype_t *, int,  void *, void *, void *, int *, struct ompi_datatype_t *)’ [-Wincompatible-pointer-types]
     copy_fn.attr_datatype_copy_fn  =
                                    ^
pml_ucx.c: In function ‘mca_pml_ucx_isend_init’:
pml_ucx.c:703:23: warning: assignment to ‘ucp_datatype_t’ {aka ‘long unsigned int’} from ‘void *’ makes integer from pointer without a cast [-Wint-conversion]
         req->datatype = NULL;
                       ^
pml_ucx.c: In function ‘mca_pml_ucx_iprobe’:
pml_ucx.c:982:64: warning: ISO C forbids conditional expr with only one void side [-Wpedantic]
             (void)ucp_worker_progress(ompi_pml_ucx.ucp_worker) : opal_progress();
                                                                ^
pml_ucx.c: In function ‘mca_pml_ucx_improbe’:
pml_ucx.c:1031:64: warning: ISO C forbids conditional expr with only one void side [-Wpedantic]
             (void)ucp_worker_progress(ompi_pml_ucx.ucp_worker) : opal_progress();
                                                                ^
pml_ucx_component.c: In function ‘mca_pml_ucx_component_register’:
pml_ucx_component.c:52:9: warning: unused variable ‘multi_send_op_attr_enable’ [-Wunused-variable]
     int multi_send_op_attr_enable;
         ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from pml_ucx_request.c:13:
pml_ucx_request.c: In function ‘mca_pml_ucx_set_recv_status’:
pml_ucx_request.h:66:14: warning: ‘tag’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     (((_tag) >> PML_UCX_CONTEXT_BITS) & UCS_MASK(PML_UCX_RANK_BITS))
              ^~
pml_ucx_request.h:183:13: note: ‘tag’ was declared here
     int64_t tag;
             ^~~
```

### PR DESCRIPTION
**Pull request by karasevb**
Fixes #9984

### CODE REVIEW

File: oshmem/mca/atomic/ucx/atomic_ucx_module.c [hunk 1]
```diff
@@ -324,14 +324,14 @@ static int mca_atomic_ucx_swap_nb(shmem_ctx_t ctx,
...
-                         size_t size,
-                         int pe)
+static int mca_atomic_ucx_cswap_nb(shmem_ctx_t ctx,
+                               void *fetch,
+                               void *target,
+                               uint64_t *prev,
+                               uint64_t cond,
+                               uint64_t value,
+                               size_t size,
+                               int pe)
```
Comment (hoopoepg) [inline oshmem/mca/atomic/ucx/atomic_ucx_module.c:334]: strange alignment

**Review by hoopoepg — APPROVED** [review]
just minor comments

File: oshmem/mca/atomic/ucx/atomic_ucx_module.c (hunk 1 above)
Comment (karasevb) [reply oshmem/mca/atomic/ucx/atomic_ucx_module.c]: fixed

**Review by hoopoepg — APPROVED** [review]

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/cc048d76221442339dcc3d1675fc5901

**Comment by karasevb** [comment]
bot:ibm:retest

**Comment by karasevb** [comment]
@brminich @yosefe

File: opal/mca/common/ucx/common_ucx.c [hunk 2]
```diff
@@ -203,21 +204,23 @@ static bool opal_common_ucx_check_device(const char *device_name, char **device_
...
     char *driver_name;
     char **list_item;
     ssize_t ret;
+    char *ib_device_name_fmt;
 
+    opal_asprintf(&ib_device_name_fmt, "%%%u[^:]%%*d", NAME_MAX-1);
     /* mlx5_0:1 */
-    ret = sscanf(device_name, "%m[^:]%*d", &ib_device_name);
+    ret = sscanf(device_name, ib_device_name_fmt, &ib_device_name);
+    free(ib_device_name_fmt);
```
Comment (yosefe) [inline opal/mca/common/ucx/common_ucx.c:216]: maybe ib_device_name_fmt could be fixed size on the stack, so no need to release

File: opal/mca/common/ucx/common_ucx.c [hunk 3]
```diff
@@ -298,17 +302,17 @@ OPAL_DECLSPEC opal_common_ucx_support_level_t opal_common_ucx_support_level(ucp_
     /* Print ucx transports information to the memory stream */
     ucp_context_print_info(context, stream);
 
+    /* "# resource 6  :  md 5  dev 4  flags -- rc_verbs/mlx5_0:1" */
+    opal_asprintf(&rsc_name_fmt,
+        "# resource %%*d : md %%*d dev %%*d flags -- %%%u[^/ \n\r]/%%%u[^/ \n\r]",
+        NAME_MAX-1, NAME_MAX-1);
```
Comment (yosefe) [inline opal/mca/common/ucx/common_ucx.c:308]: minor: space around minus operator

File: opal/mca/common/ucx/common_ucx.c [hunk 4]
```diff
@@ -298,17 +302,17 @@ OPAL_DECLSPEC opal_common_ucx_support_level_t opal_common_ucx_support_level(ucp_
     /* Print ucx transports information to the memory stream */
     ucp_context_print_info(context, stream);
 
+    /* "# resource 6  :  md 5  dev 4  flags -- rc_verbs/mlx5_0:1" */
+    opal_asprintf(&rsc_name_fmt,
```
Comment (brminich) [inline opal/mca/common/ucx/common_ucx.c:306]: can use `opal_snprintf` here as well then?

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Pull request merged**